workspaces/mcp-chat/plugins/mcp-chat/src/components/ChatPage/ChatPage.tsx

```diff
@@ -0,0 +1,129 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { useState, useRef } from 'react';
+import { useTheme } from '@mui/material/styles';
+import Box from '@mui/material/Box';
+import { Content, Page, ResponseErrorPanel } from '@backstage/core-components';
+import { ChatContainer, type ChatContainerRef } from '../ChatContainer';
+import { RightPane } from '../RightPane';
+import { useProviderStatus, useMcpServers } from '../../hooks';
+
+interface Message {
+  id: string;
+  text: string;
+  isUser: boolean;
+  timestamp: Date;
+  tools?: string[];
+  toolsUsed?: string[];
+  toolResponses?: any[];
+}
+
+export const ChatPage = () => {
+  const theme = useTheme();
+
+  const providerStatus = useProviderStatus();
+  const {
+    mcpServers,
+    error: mcpServersError,
+    handleServerToggle,
+  } = useMcpServers();
+  const [sidebarCollapsed, setSidebarCollapsed] = useState(true);
+
+  const [error, setError] = useState<string | null>(null);
+  const [messages, setMessages] = useState<Message[]>([]);
+  const chatContainerRef = useRef<ChatContainerRef>(null);
+
+  // Combine errors from different sources
+  const combinedError = error || mcpServersError;
+
+  const toggleSidebar = () => {
+    setSidebarCollapsed(!sidebarCollapsed);
+  };
+
+  const handleNewChat = () => {
+    // Cancel any ongoing request first
+    if (chatContainerRef.current) {
+      chatContainerRef.current.cancelOngoingRequest();
+    }
+
+    setError(null);
+    setMessages([]);
+  };
+
+  return (
+    <Page themeId="tool">
+      <Content noPadding>
+        <Box
+          sx={{
+            display: 'flex',
+            flexDirection: 'column',
+            height: '100vh',
+            backgroundColor: theme.palette.background.default,
+          }}
+        >
+          {/* Content Area */}
+          <Box
+            sx={{
+              display: 'flex',
+              flex: 1,
+              overflow: 'hidden',
+              position: 'relative',
+            }}
+          >
+            {combinedError ? (
+              <Box
+                sx={{
+                  display: 'flex',
+                  justifyContent: 'center',
+                  alignItems: 'center',
+                  width: '100%',
+                  height: '100%',
+                  padding: '20px',
+                }}
+              >
+                <ResponseErrorPanel
+                  error={new Error(combinedError)}
+                  defaultExpanded
+                />
+              </Box>
+            ) : (
+              <>
+                {/* Chat Container */}
+                <ChatContainer
+                  ref={chatContainerRef}
+                  sidebarCollapsed={sidebarCollapsed}
+                  mcpServers={mcpServers}
+                  messages={messages}
+                  setMessages={setMessages}
+                />
+
+                {/* Sidebar - Right Side */}
+                <RightPane
```

Comment:
My you could use a drawer instead, but your solution is working fine :+1:

https://v5.mui.com/material-ui/react-drawer/#mini-variant-drawer